android/apollo/src/com/andrew/apollo/ui/fragments/RecentFragment.java

```diff
@@ -56,35 +58,42 @@ public void onCreateContextMenu(final ContextMenu menu, final View v,
     }
 
     @Override
-    protected AlbumAdapter createAdapter() {
-        int layout;
-        if (isSimpleLayout()) {
-            layout = R.layout.list_item_normal;
-        } else if (isDetailedLayout()) {
-            layout = R.layout.list_item_detailed_no_background;
-        } else {
-            layout = R.layout.grid_items_normal;
-        }
-        return new AlbumAdapter(getActivity(), layout);
+    protected SongAdapter createAdapter() {
+        return new SongAdapter(getActivity(), R.layout.list_item_simple_image);
     }
 
     @Override
     protected String getLayoutTypeName() {
-        return PreferenceUtils.RECENT_LAYOUT;
+        return PreferenceUtils.SIMPLE_LAYOUT;
+    }
+
+    @Override
+    public void onItemClick(final AdapterView<?> parent, final View view,
+                            final int position, final long id) {
+        onSongItemClick(position);
     }
 
     @Override
-    public Loader<List<Album>> onCreateLoader(final int id, final Bundle args) {
+    public Loader<List<Song>> onCreateLoader(final int id, final Bundle args) {
         return new RecentLoader(getActivity());
     }
 
     /**
      * Sets up the list view
      */
-    protected void initListView() {
-        super.initListView();
+//    TODO: Ask about this method to @gubatron because the current adapter
```

Comment:
what does "setTouchPlay(true)" do?

If it's working as expected now, then you can just get rid of the method if all it does is call its parent.